README.md

```diff
@@ -20,3 +20,98 @@ cli-kit new myproject
 
 You can see example usage [here](https://github.com/Shopify/cli-kit-example). This app is similar to
 the one that the generator generates.
+
+## Starting a New Project
+To begin creating your first `cli-kit` application, run:
+```bash
+cli-kit new myproject
+```
+Where `myproject` is the name of the application you wish to create.  Then, you will be prompted to
+select how the project consumes `cli-kit` and `cli-ui`.  The available options are:
+- Vendor (faster execution, more difficult to update dependencies)
+- Bundler (slower execution, easier dependency management)
+
+You're now ready to write your very first `cli-kit` application!
+
+## How do `cli-kit` Applications Work?
+The executable for your `cli-kit` app is stored in the "exe" directory.  To execute the app, simply
+run:
+```bash
+./exe/myproject
+```
+
+### Folder Structure
+* `/exe/` - Location of the executables for your application.
+* `/lib/` - Location of the resources for your app (modules, classes, helpers, etc).
+    * `myproject.rb` - The main file for your application.
+    * `myproject/` - Stores the various commands/entry points.
+        * `entry_point.rb` - Is executed when the application is launched, and handles commands.
+        * `commands.rb` - Registers the various commands that your application is able to handle.
+        * `commands/` - Stores Ruby files for each command (help, new, add, etc).
+
+## Adding a New Command to your App
+### Registering the Command
+Let's say that you'd like your program to be able to handle a specific task, and you'd like to
+_register_ a new handler for the command for that task, like `todo add` to add an item to a to-do
+list.
+To do this, open `/lib/myproject/commands.rb`, where `myproject` is the name of your app.  Then, add
+a new line into the module, like this:
+```ruby
+register :Add, 'add', 'myproject/commands/add'
+```
+
+The format for this is `register` `:ActionName` `'command by user'` `'path/to/command.rb'`
+
+### Creating the Command Action
+The action for a specific command is stored in it's own Ruby file, in the `/lib/myproject/commands/`
+directory.  Here is an example of the `add` command in our previous to-do app example:
+```ruby
+require 'myproject'
+
+module MyProject
+  module Commands
+    class Add < MyProject::Command
+      def call(args, _name)
+        # command action goes here
+      end
+
+      def self.help
+        # help or instructions go here
+      end
+    end
+  end
+end
+
+```
+
+The `call(args, _name)` method is what actually runs when the `myproject add` command is executed.
+
+- **Note:** The `args` parameter represents all the arguments the user has specified.
+
+Let's say that you are trying to make a to-do list app, and would like to get an argument that the
+user has specified.  `args.first` will get the first argument.  For example:
+```ruby
+def call(args, _name)
+  list = Todo::Config.get('default', 'list') || '[]'
```

Comment:
`Todo` comes from the example to do app.  But so far you've been using myproject - stick with that.  But also, this introduces cli-kit's config, which may be useful to talk about in the readme, but as its own section before its used to explain something else.  Another example that might work well here (without even needing to change the command name):
```ruby
def call(args, _name)
  sum = args.map(&:to_i).inject(&:+)
  puts sum
end
```

Reply:
I figure we can link to config's rdoc when we create that here.

Reply:
Sure, that works.  It should still be a command that makes sense in isolation since its the only one being shown here.

Reply:
I'll use the sum example for now, once I read more about how config works and actually play around with it, I'll add more about it to the README.